routing/route_weight.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "routing/base/astar_weight.hpp"
+
+#include <iostream>
+#include <limits>
+
+namespace routing
+{
+class RouteWeight final
+{
+public:
+  RouteWeight() = default;
+
+  constexpr explicit RouteWeight(double weight) : m_weight(weight) {}
+
+  double GetWeight() const { return m_weight; }
+
+  bool operator<(RouteWeight const & rhs) const { return m_weight < rhs.m_weight; }
+
+  bool operator==(RouteWeight const & rhs) const { return !((*this) < rhs) && !(rhs < (*this)); }
+
+  bool operator!=(RouteWeight const & rhs) const { return !((*this) == rhs); }
+
+  bool operator>(RouteWeight const & rhs) const { return rhs < (*this); }
+
+  bool operator>=(RouteWeight const & rhs) const { return !((*this) < rhs); }
```

Comment:
```using namespace std::rel_ops;``` ?
Given a user-defined operator== and operator< for objects of type T, implements the usual semantics of other comparison operators.